### PR TITLE
update gloassary output to add \gls remove all \gls from defn file

### PR DESCRIPTION
**Pull request by womullan**
Also escapes & _ in all the out put these are also removed from glosarydef

### CODE REVIEW

File: bin/generateAcronyms.py [hunk 1]
```diff
@@ -461,20 +463,19 @@ def write_latex_table(acronyms, dotex=True, dorst=False, fd=sys.stdout):
     for acr, defn in acronyms:
         acr = acr.replace("&", r"\&")
         acr = acr.replace("_", r"\_")
-        defn = glsreg.sub(glsrmfn, defn[0])
```
Comment (timj) [inline bin/generateAcronyms.py:464]: Removing this requires that glsreg is removed as well from a couple of lines above.

File: bin/generateAcronyms.py [hunk 2]
```diff
@@ -576,6 +579,27 @@ def main(texfiles, doGlossary, utags, dotex, dorst, mode):
     return len(results)
 
 
+def update_gls_entries(results, GLSlist):
+    """ Scan through the acronym anf gls definitions and add \gls where
```
Comment (timj) [inline bin/generateAcronyms.py:583]: typo: anf

File: bin/generateAcronyms.py [hunk 3]
```diff
@@ -593,10 +617,33 @@ def glsfn(s):
...
+    """ Make a re map of regexps for substitution"""
+    regexmap = {}
+    for g in GLSlist:
+        if g != '':
+            regexmap[g] = re.compile(r"([,\s(](?<!={))("+g+r")(\b)")
+    return regexmap
+
+def sub_line (line, regexmap, GLSlist):
+    """ for give line put \gls around gls items not already adorned.
+    :param line:
```
Comment (timj) [inline bin/generateAcronyms.py:630]: Either use numpydoc following DM convention, or remove the `:param:` lines completely (as written they are just repeating the arguments anyhow).

File: bin/generateAcronyms.py [hunk 4]
```diff
@@ -593,10 +617,33 @@ def glsfn(s):
...
 
-
-def glsrmfn(s):
-    """remove \\gls{} -- used in the regexp substitution for acronym"""
-    return s.group(1)
+def make_regexmap (GLSlist):
+    """ Make a re map of regexps for substitution"""
+    regexmap = {}
+    for g in GLSlist:
+        if g != '':
```
Comment (timj) [inline bin/generateAcronyms.py:624]: Is there a reason that `if g:` doesn't work?

File: bin/generateAcronyms.py [hunk 5]
```diff
@@ -593,10 +617,33 @@ def glsfn(s):
...
+            regexmap[g] = re.compile(r"([,\s(](?<!={))("+g+r")(\b)")
+    return regexmap
+
+def sub_line (line, regexmap, GLSlist):
+    """ for give line put \gls around gls items not already adorned.
```
Comment (timj) [inline bin/generateAcronyms.py:629]: This needs to be an r-string because of that `\gls`.

give -> given

File: bin/generateAcronyms.py [hunk 6]
```diff
@@ -593,10 +617,33 @@ def glsfn(s):
...
+    :param line:
+    :param regexmap:
+    :param GLSlist:
+    :return: modifed line
+    """
+    nline = line
+    for g in GLSlist:
+        regx = regexmap[g]
+        res = regx.search(line)
+        if (res is not None):  # ok now .. more checks
```
Comment (timj) [inline bin/generateAcronyms.py:639]: The parentheses are not needed

File: bin/generateAcronyms.py [hunk 7]
```diff
@@ -593,10 +617,33 @@ def glsfn(s):
...
+        if g != '':
+            regexmap[g] = re.compile(r"([,\s(](?<!={))("+g+r")(\b)")
+    return regexmap
+
+def sub_line (line, regexmap, GLSlist):
+    """ for give line put \gls around gls items not already adorned.
+    :param line:
+    :param regexmap:
+    :param GLSlist:
+    :return: modifed line
```
Comment (timj) [inline bin/generateAcronyms.py:633]: modifed to modified

File: bin/generateAcronyms.py [hunk 8]
```diff
@@ -605,25 +652,13 @@ def updateFile(inFile, GLSlist):
...
-                        res = regx.search(line)
-                        if (res is not None):  # ok now .. more checks
-                            # check its not a word in a GLS item but not
-                            # too greedy
-                            glsed = re.search(r"gls{.+"+g+"[a-z,A-Z, ]*}", line)
-                            if (glsed):  # already glsed or contained in one
-                                continue
-                            else:  # .. find and add GLS -
-                                line = regx.sub(glsfn, line)
+                    sub_line(line,regexmap,GLSlist)
```
Comment (timj) [inline bin/generateAcronyms.py:661]: I'm a bit confused because this isn't returning anything so what is the sub_line meant to do?

**Review by timj — APPROVED** [review]
I don't understand what sub_line is doing in one place.

Lots of flake8 problems need fixing.

File: bin/generateAcronyms.py (hunk 1 above)
Comment (womullan) [reply bin/generateAcronyms.py:464]: ok .. its not removed only refactored ..

Comment (timj) [reply bin/generateAcronyms.py:464]: My mistake. I thought the flake8 report said it was no longer used.

File: bin/generateAcronyms.py (hunk 2 above)
Comment (womullan) [reply bin/generateAcronyms.py:586]: ok

File: bin/generateAcronyms.py (hunk 3 above)
Comment (womullan) [reply bin/generateAcronyms.py:633]: ok

File: bin/generateAcronyms.py (hunk 5 above)
Comment (womullan) [reply bin/generateAcronyms.py:632]: ok

File: bin/generateAcronyms.py (hunk 6 above)
Comment (womullan) [reply bin/generateAcronyms.py:642]: old habits hard to break

File: bin/generateAcronyms.py (hunk 8 above)
Comment (womullan) [reply bin/generateAcronyms.py:664]: good catch total error !

File: bin/generateAcronyms.py (hunk 7 above)
Comment (womullan) [reply bin/generateAcronyms.py:636]: ok

File: bin/generateAcronyms.py (hunk 4 above)
Comment (womullan) [reply bin/generateAcronyms.py:627]: actually i think this is not needed - there was a ,,,,, in the definition file giving a blank entry in the definitions ..

File: bin/generateAcronyms.py (hunk 1 above)
Comment (womullan) [reply bin/generateAcronyms.py:464]: ok mistakenly removed ..

**Comment by womullan** [comment]
ok flake fixed and the one mistake with sub_line

**Pull request merged**